### ISSUE TITLE
Stopping by index terminates multiple processes (0.14.1)

### ISSUE THREAD

**Issue by Kegsay** [foreverjs/forever#699]
```
$ forever --version
v0.14.1

$ forever list
info:    Forever processes running
data:        uid  command         script forever pid   id logfile                       uptime       
data:    [0] yinz /usr/bin/nodejs app.js 13778   13780    /home/kegsay/.forever/yinz.log 0:0:4:3.199  
data:    [1] kABa /usr/bin/nodejs app.js 13801   13803    /home/kegsay/.forever/kABa.log 0:0:3:45.139

$ forever stop 0
info:    Forever stopped process:
    uid  command         script forever pid   id logfile                       uptime       
[0] yinz /usr/bin/nodejs app.js 13778   13780    /home/kegsay/.forever/yinz.log 0:0:4:12.408 
[1] kABa /usr/bin/nodejs app.js 13801   13803    /home/kegsay/.forever/kABa.log 0:0:3:54.348

$ forever list
info:    No forever processes running
```

Expected: That the first process ``[0]`` would be killed by ``forever stop 0``.
Actual: Both processes were terminated.

This may be related to: https://github.com/foreverjs/forever/issues/481
Running node ``v0.10.25``

**Comment by Kegsay** [comment]
This doesn't appear to be based on having the same name such as ``app.js``. This only seems to happen if you try to ``forever stop 0``. In other words, ``forever stop 1`` works correctly (stopping the 2nd process marked ``[1]`` in the list).

**Comment by Kegsay** [comment]
I suspect the bug is at https://github.com/foreverjs/forever/blob/509eaf24deaf1ea8515298d5fcf16708715e658c/lib/forever.js#L692 which is doing an ``==`` check when I think it really needs ``===``. The variables when reproducing this bug are:
```
> p.id
false
> id
0
```

**Comment by nameanyone** [comment]
Related to  #659?

**Comment by danzaner** [comment]
@Kegsay  I can confirm that your fix solves this problem for me. I consistently had all processes stopped when issuing a 'forever stop 0'. Modifying forever.findById() from 'return p.id == id;' to 'return p.id === id;' results in 'forever stop 0' killing only the process at index 0.

Many thanks!

**Comment by devinivy** [comment]
We're also experiencing this issue.

**Comment by indexzero** [comment]
Fixed in 0.14.2

**Comment by nameanyone** [comment]
I have 0.14.2, and `forever stop 0` still stops everything.

**Comment by devinivy** [comment]
Sounds like this simply needs a failing test.  Does anyone mind writing one?

**Comment by ultimate-tester** [comment]
I can confirm that the latest NPM version does not contain the fix yet! Code of the affected file does not match the file in master branch.

**Comment by playeren** [comment]
This issue still exists in v0.15.1 where #756 is fixed.
 
```
root@freja:/srv/freshHQ/main# forever --version
v0.15.1

root@freja:/srv/freshHQ/main# forever restart 0
info:    Forever restarted process(es):
data:        uid  command                                  script    forever pid   id logfile                 uptime
data:    [0] ZX72 /root/.nvm/versions/node/v6.9.1/bin/node websrv.js 28449   28455    /root/.forever/ZX72.log 0:0:0:15.776
data:    [1] OoP3 /root/.nvm/versions/node/v6.9.1/bin/node btcapi.js 28475   28481    /root/.forever/OoP3.log 0:0:0:8.245
```

**Comment by Kegsay** [comment]
I don't think #756 landed in 0.15.1:
 - 31 Jul 2015 : Bumped to 0.15.1 : https://github.com/foreverjs/forever/commit/6c157fafe909eb555b9e73028338377b05d51f51
 - 29 Aug 2015 : Fix landed https://github.com/foreverjs/forever/commit/a957354623cf1de189ee277eb16262f832409b1b
 - 26 May 2016 : Bumped to 0.15.2 https://github.com/foreverjs/forever/commit/66e8c0b72aeaa43f814f99ef4bc412db717ebd97
